fragments/spring-boot-app-sso-auth-code-flow/src/main/java/com/vmware/tap/fragments/appssoauthcodeflow/WebSecurityConfiguration.java

```diff
@@ -0,0 +1,55 @@
+package com.vmware.tap.fragments.appssoauthcodeflow;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.oauth2.client.oidc.web.logout.OidcClientInitiatedLogoutSuccessHandler;
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.logout.LogoutSuccessHandler;
+
+import static org.springframework.security.config.Customizer.withDefaults;
+
+@EnableWebSecurity
+public class WebSecurityConfiguration {
+
+    @Autowired
+    private ClientRegistrationRepository clientRegistrationRepository;
+
+    @Bean
+    SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+        http
+                .authorizeRequests(authorizeRequests ->
+                        authorizeRequests
+                                // Permit all public paths explicitly
+                                .antMatchers("/",
+                                        "/livez",
+                                        "/readyz",
+                                        "/webjars/**",
+                                        "/styles/**",
+                                        "/images/**",
+                                        "/favicon.ico").permitAll()
+                                // Secure our Spring MVC Endpoints (basically the rest)
+                                .anyRequest().authenticated()
+                )
+                // After a successful logout, redirect to /.
+                .logout().logoutSuccessHandler(oidcLogoutSuccessHandler()).logoutSuccessUrl("/")
+                .and()
```

Comment:
This should be using the lambda configuration methods instead of chaining with `.and()`.
```suggestion
                .logout(logout -> logout
                    .logoutSuccessHandler(oidcLogoutSuccessHandler())
                    .logoutSuccessUrl("/")
                )
```